src/pages/docs/guides/pubsub/push-notifications.mdx

```diff
@@ -0,0 +1,526 @@
+{
+  '*': ['subscribe'],                        // Can subscribe to realtime messages
+  'news:*': ['subscribe', 'push-subscribe']  // Can also subscribe to push notifications on news channels
+}
+
+// For your server with push-admin
+{
+  '*': ['subscribe', 'publish', 'push-admin']  // Full push admin access
+}
+```
+</Code>
+
+Capabilities can be set for API keys, Ably tokens, and JWT tokens. Read more about [authentication with Ably](/docs/auth).
+
+If you are using channel-based push notifications, be aware that everyone with `publish` capability can publish messages to that channel, including sending push notifications. We recommend that only your server has `publish` capability on channels you want to use for push notifications.
+
+### Security considerations
+
+Follow these practices:
+
+* Never give clients `push-admin` capability. They could manage registrations and subscriptions for push notifications for any device.
+* Do not give `publish` capability to clients on channels used for push notifications. This could allow any clients to publish push notifications to that channel.
+* To make permission management straightforward, use separate channels for push notifications and other parts of your application.
+* Only grant the capabilities that are needed for the specific use case.
+* Use token authentication and not API key authentication.
+
+## Managing device lifecycle
+
+Device management is critical for a healthy push notification system. Devices don't last forever: users upgrade phones, uninstall apps, revoke notification permissions, or log out and in with different accounts.
+
+Devices need updates when:
+
+* The user logs in or out (changing `clientId`).
+* The platform token changes (for example, FCM tokens refresh).
+* The user disables push notifications for the app and later re-enables them.
+
+<Code>
+```javascript
+// Update device registration
+await client.push.admin.deviceRegistrations.save({
+  id: 'device-123',
+  clientId: 'new-user-id',  // User logged into different account
+});
+```
+</Code>
+
+### Handling deactivation
+
+When users disable notifications in your app or log out, remove the device registration from Ably:
+
+<Code>
+```javascript
+// User disables notifications in app settings
+await client.push.deactivate();
+```
+</Code>
+
+The device will not receive further notifications until reactivated.
+
+## Subscription management at scale
+
+If using channel-based notifications, managing subscriptions efficiently is important as your user base grows.
+
+### Channel organization
+
+Good channel design makes it easy for users to subscribe to exactly what they want, and for you to target notifications precisely. Design your channels thoughtfully:
+
+* General categories: topic, geography, user segment.
+* Personalized user interests: groups the user is in, discussion threads or chat rooms the user is active in, documents the user is working on, and more.
+* Any other business logic.
+
+<Code>
+```javascript
```

Comment:
I wouldn't put this as a code block - as otherwise we have to duplicate across each (and it isn't code).